tests/phpunit/tests/widgets.php

```diff
@@ -669,7 +670,9 @@ public function test_wp_widget_get_settings() {
 		$never_used = get_option( 'widget_nav_menu', array() );
 		$this->assertSame( array(), (array) $never_used );
 
+		$num_queries_before_init = get_num_queries();
 		wp_widgets_init();
+		$this->assertEquals( $num_queries_before_init + 1, get_num_queries() ); // only 1 query for widget init.
 		$wp_widget_block = $wp_registered_widgets['block-2']['callback'][0];
 
 		$settings = $wp_widget_block->get_settings();
```

Comment:
@uday-kokitkar shouldn't you check the number of queries after get_settings? as far as I see in your changes, this is when using an alt_option_name, should save a query.